hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -535,6 +537,11 @@ private HoodieWriteConfig getHoodieClientConfig(SchemaProvider schemaProvider) {
     }
     HoodieWriteConfig config = builder.build();
 
+    // set default value for {@link HoodieWriteCommitKafkaCallbackConfig} if needed.
+    if (config.writeCommitCallbackOn() && HoodieWriteCommitKafkaCallback.class.getName().equals(config.getCallbackClass())) {
```

Comment:
If users configure http callback? Why we only consider Kafka callback here?

Reply:
> 
> 
> If users configure http callback? Why we only consider Kafka callback here?

the config of http callback will be Initialized when HoodieWriteClient set up, while kafka's can't. the kafka's config can only be set up in hudi-utilities module, so the logic above come up.